src/EditorFeatures/CSharpTest/Diagnostics/SpellCheck/SpellCheckTests.cs

```diff
@@ -461,5 +461,49 @@ void M()
 
             await TestAsync(text, expected, compareTokens: false);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsSpellcheck)]
+        [WorkItem(13345, "https://github.com/dotnet/roslyn/issues/13345")]
+        public async Task TestMissingOnSnippet()
+        {
+            await TestAsync(
```

Comment:
I can have another test, but this test is to verify this specific behavior.  Namely, 'for' in a context like ```var i = for``` only shows up because it's a snippet.  However, ```for``` at the beginning of a statement shows up both because it's a keyword and a snippet. So in the former case, we don't want to offer to spell check to it, but the latter case makes sure we're not just filtering out all keywords.

Without htis test, it would be possible to mistakingly implement the feature just by filtering out snippet items.  But that would make us not offer to spell check in *either* case.  That's because the completion service always overrides items with snippet items.  This test ensures that if we pass along the right option that we don't even produce hte snippets that would override hte keywords.